src/Control/Effect/Resource.hs

```diff
@@ -14,14 +15,17 @@ import           Control.Monad.IO.Class
 
 data Resource m k
   = forall resource any output . Resource (m resource) (resource -> m any) (resource -> m output) (output -> k)
+  | forall resource any output . OnError  (m resource) (resource -> m any) (resource -> m output) (output -> k)
```

Comment:
I figured that this is the most logical choice for this constructor name. I would prefer at this point to rename `Resource` to `Bracket`, since `Resource` is no longer a single-constructor type, but that would break backwards compatibility (though we are already doing that next release, FWIW).